pkg/controller/controller_instance.go

```diff
@@ -1103,13 +1110,18 @@ func isServiceInstanceProcessedAlready(instance *v1beta1.ServiceInstance) bool {
 // processServiceInstancePollingFailureRetryTimeout marks the instance as having
 // failed polling due to its reconciliation retry duration expiring
 func (c *controller) processServiceInstancePollingFailureRetryTimeout(instance *v1beta1.ServiceInstance, readyCond *v1beta1.ServiceInstanceCondition) error {
+	msg := "Stopping reconciliation retries because too much time has elapsed"
```

Comment:
send the event here?

Reply:
The event is being sent inside `processServiceInstancePollingTerminalFailure` method invoked 2 lines below.

Reply:
Or rather in functions invoked by `processServiceInstancePollingTerminalFailure`... :)

Reply:
it all looks very spaghetti :-\

Reply:
@artkoshelev You're welcome to submit a PR to refactor this for readability! 😀 Nile did the right thing by sticking with fixing just the bug in this PR.